subcommands/keys/tuf_updates_sign_prod_targets.go

```diff
@@ -65,12 +71,25 @@ For example, add a new offline TUF targets key, before signing production target
 		subcommands.SliceRemove(newCiRoot.Signed.Roles["targets"].KeyIDs, onlineTargetsId))
 	subcommands.DieNotNil(err)
 
+	var newTargetsProdSigs, newTargetsWaveSigs map[string][]tuf.Signature
+
 	fmt.Println("= Signing prod targets")
-	targetsMap, err := api.ProdTargetsList(factory, true, tags...)
-	subcommands.DieNotNil(err, "Failed to fetch production targets:")
-	newTargetsSigs, err := signProdTargets(signer, targetsMap)
-	subcommands.DieNotNil(err)
+	// If both wave names and tags specified, or none specified - re-sign both prod and wave targets.
+	// If only wave names or only tags specified - re-sign only what was specified (either wave names or tags).
+	if len(tags) > 0 || len(waveNames) == 0 {
+		targetsProdMap, err := api.ProdTargetsList(factory, true, tags...)
+		subcommands.DieNotNil(err, "Failed to fetch production targets:")
+		newTargetsProdSigs, err = signProdTargets(signer, targetsProdMap)
+		subcommands.DieNotNil(err)
+	}
+	if len(waveNames) > 0 || len(tags) == 0 {
+		targetsWaveMap, err := api.WaveTargetsList(factory, true, waveNames...)
+		subcommands.DieNotNil(err, "Failed to fetch production wave targets:")
```

Comment:
just a nit, not sure `production` is needed in the error message.

Reply:
what will happen if both `tags` and `waves` are not specified?

Reply:
I found it is useful.

Although `wave targets` implicitly presumes `production` attribute; I indeed found it useful to use both words in this context.

Reply:
> what will happen if both tags and waves are not specified?

The default action will happen, which is to sign all production targets and all active waves, using the specified targets key.

Reply:
I suspect the majority of users will only ever run the default command.

Reply:
what will happen if both tags and waves are not specified?